README.md

```diff
@@ -446,3 +449,17 @@ Then set it up:
 export BTCPAYGEN_ADDITIONAL_FRAGMENTS="opt-save-storage.custom"
 . ./btcpay-setup.sh -i
 ```
+
+## Can I run BTCPay Server on ports other than 80 and 443?
+
+You can change the ports for HTTP and HTTPS by setting the environment variables `REVERSEPROXY_HTTP_PORT` and `REVERSEPROXY_HTTPS_PORT`. This is handy when ports 80 and 443 are already in use on your host, or you want to offload SSL termination with an existing web proxy.
+
+When you set `REVERSEPROXY_HTTP_PORT` to another value than 80, the built-in Let's Encrypt certificate will not work, as Let's Encrypt will try to validate your SSL certificate request by connecting from the internet to your domain on port 80. This validation request should be able to reach BTCPay Server in order to receive the certificate.
+
+If you need to run on a different port, it's best to terminate SSL using another web proxy and foreard your traffic. 
+
+## Can I offload HTTPS termination? 
+
+Yes. To offload SSL termination, just forward the requests to the port specified by `REVERSEPROXY_HTTP_PORT` and make sure you are setting the header `X-Forwarded-Proto: https` so BTC Pay Server can know the original request was HTTPS. If you forget this extra header, BTCPay Server will work, but it will believe the connection is insecure and display a warning message.
```

Comment:
Actually, this is not enough, see https://docs.btcpayserver.org/faq-and-common-issues/faq-deployment#btcpay-is-expecting-you-to-access-this-website-from

I advise you to link to the doc.

Reply:
I read that doc, but IMHO there is no need to edit the original `/etc/nginx/conf.d/default.conf`.

The `map{}` directive isn't needed as you can use the built-in `$scheme` var instead.

This is the vhost config I currently use. Everything else is default Nginx on Debian Stretch.
```
cat /etc/nginx/sites-available/btcpayserver 

server {
	listen 80;

        root /var/www/html;
        index index.html index.htm index.nginx-debian.html;

        server_name yourdomain.com;

        location ~ /.well-known {
                allow all;
        }

        location / {
		return 301 https://yourdomain.com$request_uri;
        }
}

server {
        listen 443 ssl http2;

        ssl on;
        ssl_certificate      /etc/letsencrypt/live/yourdomain.com/fullchain.pem;
        ssl_certificate_key  /etc/letsencrypt/live/yourdomain.com/privkey.pem;

        root /var/www/html;
        index index.html index.htm index.nginx-debian.html;

        server_name yourdomain.com;

        location / {
                proxy_pass http://127.0.0.1:1080;

                proxy_set_header Host $http_host;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        }

        location ~ /.well-known {
                allow all;
        }
}
```

Agreed, more testing may be needed, but I'm able to access the site, backend, etc. Haven't done any heavy production use though.

Reply:
I'm not sure if these additional settings from the doc are actually needed.
Would be nice to see more comments on why they're there:
```
server_names_hash_bucket_size 128;
proxy_buffer_size          128k;
proxy_buffers              4 256k;
proxy_busy_buffers_size    256k;
http2_max_field_size       32k;
```

Reply:
Sorry, yes.
So when you use the internal wallet, we pass PSBTs via request query `?psbt=blahblah`.
Problem is that if psbt is too long then you get a Connection Error.

I traced back this connection error on NGinx which serve HTTP2 content, even if the upstream is using HTTP. And I noticed those settings fixed the issue.

For `server_names_hash_bucket_size`, it is to allow long domain name without crash

Reply:
OK, this makes sense indeed. Can we update the page at https://docs.btcpayserver.org/faq-and-common-issues/faq-deployment#btcpay-is-expecting-you-to-access-this-website-from ?

The `map{}` part can be left out (it's a very confusing construction if you ask me, and not needed thanks to `$scheme`) and maybe include my vhost sample? It's cleaner.

So the user needs to do 2 things: copy-paste the proxy vars + create a vhost file in `sites-available` (always a good idea) + create a symlink to it in `sites-enabled`.

This doc section is for advanced users, so I don't think we should mind the extra complexity. Just follow best practises. The first time I read that doc, even I was confused by the single step to edit `/etc/nginx/conf.d/default.conf`. It just doesn't make sense if your Nginx is running more than 1 app.

Reply:
Sure, please can you do a PR there you are more up to date to the best practices than I am.